chain/registry.go

```diff
@@ -0,0 +1,38 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package chain
+
+import "github.com/ava-labs/hypersdk/codec"
+
+type Registry interface {
+	ActionRegistry() *codec.TypeParser[Action]
+	AuthRegistry() *codec.TypeParser[Auth]
+	OutputRegistry() *codec.TypeParser[codec.Typed]
+}
```

Comment:
1. Why do we need this interface? It feels like this could be a simple struct that just wraps the individual codecs instead of an interface
2. Why even couple the codecs at all w/ a wrapper type instead of passing around each codec individually?